domain_access/domain_access.module

```diff
@@ -25,21 +26,24 @@ define('DOMAIN_ACCESS_USER_FIELD', 'field_domain_user');
 /**
  * Implements hook_node_grants().
  */
-function domain_access_node_grants($op, AccountInterface $account) {
+function domain_access_node_grants(AccountProxy $account, $op) {
```

Comment:
Looks like we want AccountInterface not AccountProxy.

https://api.drupal.org/api/drupal/core!modules!node!node.api.php/function/hook_node_grants/8